test/integration/isolated/examples_httproute_test.go

```diff
@@ -75,6 +86,201 @@ func TestHTTPRouteWithBrokenPluginFallback(t *testing.T) {
+		WithSetup("deploy kong addon into cluster", featureSetup(
+			withControllerManagerOpts(
+				helpers.ControllerManagerOptAdditionalWatchNamespace(namespace),
+				helpers.ControllerManagerOptFlagUseLastValidConfigForFallback(),
+			),
+			withControllerManagerFeatureGates(map[string]string{featuregates.FallbackConfiguration: "true"}),
+		)).
+		Assess("deploying to cluster works and HTTP requests are routed properly", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			runHTTPRouteExampleTestScenario(httprouteExampleManifest)(ctx, t, c)
+
+			t.Log("getting a gateway client")
+			gatewayClient, err := gatewayclient.NewForConfig(GetClusterFromCtx(ctx).Config())
+			assert.NoError(t, err)
+			ctx = SetInCtxForT(ctx, t, gatewayClient)
+
+			t.Log("adding additional properly configured route")
+			_, err = gatewayClient.GatewayV1().HTTPRoutes(namespace).Create(ctx, &gatewayapi.HTTPRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      additionalRouteName,
+					Annotations: map[string]string{
+						annotations.AnnotationPrefix + annotations.StripPathKey: "true",
+					},
+				},
+				Spec: gatewayapi.HTTPRouteSpec{
+					CommonRouteSpec: gatewayapi.CommonRouteSpec{
+						ParentRefs: []gatewayapi.ParentReference{
+							{
+								Name: "kong",
+							},
+						},
+					},
+					Rules: []gatewayapi.HTTPRouteRule{
+						{
+							Matches: []gatewayapi.HTTPRouteMatch{
+								{
+									Path: &gatewayapi.HTTPPathMatch{
+										Type:  lo.ToPtr(gatewayapi.PathMatchPathPrefix),
+										Value: lo.ToPtr(additionalRoutePath),
+									},
+								},
+							},
+							BackendRefs: []gatewayapi.HTTPBackendRef{
+								{
+									BackendRef: gatewayapi.BackendRef{
+										BackendObjectReference: gatewayapi.BackendObjectReference{
+											Kind: lo.ToPtr(gatewayapi.Kind("Service")),
+											Name: additionalRoutServiceTarget,
+											Port: lo.ToPtr(gatewayapi.PortNumber(80)),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}, metav1.CreateOptions{})
+			assert.NoError(t, err)
+
+			t.Logf("verifying that routing to %s works", additionalRoutePath)
+			proxyURL := GetHTTPURLFromCtx(ctx)
+			helpers.EventuallyGETPath(
+				t,
+				proxyURL,
+				proxyURL.Host,
+				additionalRoutePath,
+				http.StatusOK,
+				additionalRoutServiceTarget,
+				nil,
+				consts.IngressWait,
+				consts.WaitTick,
+			)
+
+			return ctx
+		}).
+		Assess("assign broken broken plugin to a working route", func(ctx context.Context, t *testing.T, _ *envconf.Config) context.Context {
+			cluster := GetClusterFromCtx(ctx)
+
+			client, err := clientset.NewForConfig(cluster.Config())
+			require.NoError(t, err)
+			brokenPlugin := &kongv1.KongPlugin{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      "response-transformer",
+				},
+				PluginName: "response-transformer",
+				// Misconfigured on purpose.
```

Comment:
Let's first create a route with working KongPlugin to ensure that the last valid version of a plugin is used once we break it.